### PR TITLE
Consolidate filtering options

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-2044

### CODE REVIEW

File: docs/modules/ROOT/pages/configuration.adoc [hunk 1]
```diff
@@ -72,6 +72,10 @@ A "rule name:term"::
 This is similar to "package name:term", but allows finer granularity by only
 applying to a particular policy rule within the package.
 
+A "collection name"::
+
+Specify a predefined collection.
```
Comment (simonbaird) [inline docs/modules/ROOT/pages/configuration.adoc:77]: Should the @ prefix be mentioned here in some way?

File: internal/evaluator/conftest_evaluator.go [hunk 2]
```diff
@@ -487,31 +487,27 @@ func isResultEffective(failure output.Result, now time.Time) bool {
...
-			collections = cfg.Collections
-		}
-		if len(cfg.Include) > 0 {
-			includes = cfg.Include
-		}
-		if len(cfg.Exclude) > 0 {
-			excludes = cfg.Exclude
+		for _, c := range cfg.Collections {
+			// If the old way of specifying collections are used, convert them.
+			includes = append(includes, "@"+c)
```
Comment (simonbaird) [inline internal/evaluator/conftest_evaluator.go:497]: Oooh nice

File: internal/evaluator/conftest_evaluator.go [hunk 3]
```diff
@@ -487,31 +487,27 @@ func isResultEffective(failure output.Result, now time.Time) bool {
...
 		}
+		includes = append(includes, cfg.Include...)
+		excludes = append(excludes, cfg.Exclude...)
 	}
 
-	if len(includes)+len(collections) == 0 {
+	if len(includes) == 0 {
```
Comment (simonbaird) [inline internal/evaluator/conftest_evaluator.go:503]: This diff feels good.

File: internal/evaluator/conftest_evaluator.go [hunk 4]
```diff
@@ -487,31 +487,27 @@ func isResultEffective(failure output.Result, now time.Time) bool {
...
 		includes = []string{"*"}
 	}
 
-	include := scoreMatches(ruleMatchers, includes) + scoreMatches(collectionMatchers, collections)
-	exclude := scoreMatches(ruleMatchers, excludes)
+	includeScore := scoreMatches(ruleMatchers, includes)
+	excludeScore := scoreMatches(ruleMatchers, excludes)
 
-	return include > exclude
+	return includeScore > excludeScore
```
Comment (simonbaird) [inline internal/evaluator/conftest_evaluator.go:510]: Just noting: Exclude wins if all else is equal. That seems like what we'd want.

**Review by simonbaird — APPROVED** [review]
:rocket:

File: docs/modules/ROOT/pages/configuration.adoc (hunk 1 above)
Comment (lcarva) [reply docs/modules/ROOT/pages/configuration.adoc]: Yes, good catch. Added.

**Review by zregvart — APPROVED** [review]
Neat :+1:

**Pull request merged**